### PR TITLE
Add Moderations API

### PR DESCRIPTION
**Pull request by rukolahasser**
Implementation of `moderations` api

### CODE REVIEW

**Review by valentinegb — CHANGES_REQUESTED** [review]
Make structure fields public, currently inaccessible outside of `moderations` module.

File: src/moderations.rs [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+use derive_builder::Builder;
+use serde::{Deserialize, Serialize};
+
+use super::{ApiResponseOrError, openai_post};
+
+#[derive(Deserialize, Clone, Debug)]
+pub struct Moderation {
+    pub id: String,
+    #[serde(rename = "results")]
```
Comment (valentinegb) [inline src/moderations.rs:11]: Remove unnecessary field rename.

```suggestion
```

**Review by valentinegb — CHANGES_REQUESTED** [review]

File: src/moderations.rs (hunk 1 above)
Comment (rukolahasser) [reply src/moderations.rs]: Fixed

**Comment by rukolahasser** [comment]
Changed all visibility modifiers to `pub`

**Review by valentinegb — CHANGES_REQUESTED** [review]

**Comment by valentinegb** [comment]
Probably the last change requests haha, sorry-

File: src/moderations.rs [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+pub struct Moderation {
+    pub id: String,
+    pub results: Vec<ModerationResult>,
+}
+
+#[derive(Deserialize, Clone, Debug)]
+pub struct ModerationResult {
+    pub categories: Categories,
+    pub category_scores: CategoryScores,
+    pub flagged: bool,
```
[the remark this answers is not in the document]
Comment (rukolahasser) [reply src/moderations.rs:18]: Could you tell me where to check the schema? So my order is based on the example response:
```
{
  "id": "modr-5MWoLO",
  "model": "text-moderation-001",
  "results": [
    {
      "categories": {
        "hate": false,
        "hate/threatening": true,
        "self-harm": false,
        "sexual": false,
        "sexual/minors": false,
        "violence": true,
        "violence/graphic": false
      },
      "category_scores": {
        "hate": 0.22714105248451233,
        "hate/threatening": 0.4132447838783264,
        "self-harm": 0.005232391878962517,
        "sexual": 0.01407341007143259,
        "sexual/minors": 0.0038522258400917053,
        "violence": 0.9223177433013916,
        "violence/graphic": 0.036865197122097015
      },
      "flagged": true
    }
  ]
}

```

File: src/moderations.rs [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    pub hate_or_threatening: f64,
+    #[serde(rename = "self-harm")]
+    pub self_harm: f64,
+    pub sexual: f64,
+    #[serde(rename = "sexual/minors")]
+    pub sexual_or_minors: f64,
+    pub violence: f64,
+    #[serde(rename = "violence/graphic")]
+    pub violence_or_graphic: f64,
+}
```
[the remark this answers is not in the document]
Comment (rukolahasser) [reply src/moderations.rs]: fixed.

File: src/moderations.rs [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+#[derive(Deserialize, Clone, Debug)]
+pub struct ModerationResult {
+    pub categories: Categories,
+    pub category_scores: CategoryScores,
+    pub flagged: bool,
+}
+
+#[derive(Deserialize, Clone, Debug)]
```
[the remark this answers is not in the document]
Comment (rukolahasser) [reply src/moderations.rs]: fixed.

File: src/moderations.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (valentinegb) [reply src/moderations.rs:19]: The schema is in this repository owned by OpenAI: https://github.com/openai/openai-openapi/blob/master/openapi.yaml

[the remark this answers is not in the document]
Comment (rukolahasser) [reply src/moderations.rs]: Gotcha, fixed!

**Comment by rukolahasser** [comment]
It was closed by accident, I've created a new one: https://github.com/valentinegb/openai/pull/54

**Pull request closed without merging**